moonshot/src/utils/log.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import os
+from pathlib import Path
+
+
+def configure_logger(name: str):
+    """
+    Configures and returns a logger with a specified name.
+
+    This function creates a logger and sets its level to INFO. It also creates a console handler,
+    sets its level to INFO, and assigns a specific formatter to it. The formatter includes the
+    timestamp, log level, filename, function name, line number, and the log message. Finally, the
+    console handler is added to the logger.
+
+    Args:
+        name (str): The name of the logger to be created and configured.
+
+    Returns:
+        logging.Logger: The configured logger with the specified name.
+    """
+    log_extension = ".log"
+
+    # Read environment variable
+    log_filename = os.getenv("MS_LOG_NAME", "moonshot").lower()
+    log_level = os.getenv("MS_LOG_LEVEL", "INFO").upper()
+    log_write_to_file = os.getenv("MS_LOG_TO_FILE", "false").lower() == "true"
+
+    # Create a formatter
+    formatter = logging.Formatter(
+        "%(asctime)s [%(levelname)s][%(filename)s::%(funcName)s(%(lineno)d)] %(message)s"
+    )
+
+    # Check name is valid
+    if not name or not isinstance(name, str) or name is None:
+        name = Path(__file__).stem
+
+    # Check log level is valid
+    valid_log_levels = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
```

Comment:
is critical planned for the future when someone uses Moonshot in a production env?

Reply:
okay, we will remove the critical first.